### PR TITLE
Add support to extend the client metrics reporting

### PR DESCRIPTION
**Pull request by yma88**

### CODE REVIEW

File: honeycomb/core/src/main/java/org/commonjava/o11yphant/honeycomb/HoneycombFilter.java [hunk 1]
```diff
@@ -68,6 +69,7 @@ public void doFilter( final ServletRequest request, final ServletResponse respon
             if ( rootSpan != null )
             {
                 rootSpan.addField( "path_info", hsr.getPathInfo() );
+                addHeaderFields( hsr );
```
Comment (ruhan1) [inline honeycomb/core/src/main/java/org/commonjava/o11yphant/honeycomb/HoneycombFilter.java:72]: can you add the path_info field by the new method as well?

File: honeycomb/core/src/main/java/org/commonjava/o11yphant/honeycomb/DefaultHoneycombManager.java [hunk 2]
```diff
@@ -69,6 +72,17 @@ public DefaultHoneycombManager( HoneycombConfiguration honeycombConfiguration,
         }
     }
 
+    public DefaultHoneycombManager( HoneycombConfiguration honeycombConfiguration,
+                                    DefaultTraceSampler defaultTraceSampler )
+    {
+        super( honeycombConfiguration, defaultTraceSampler );
+    }
+
+    public void addSpanFields ( List<RootSpanFields> fields )
```
Comment (ruhan1) [inline honeycomb/core/src/main/java/org/commonjava/o11yphant/honeycomb/DefaultHoneycombManager.java:81]: pls use the registerRootSpanFields in the super class. which is just for cases when CDI is not provided.

File: honeycomb/core/src/main/java/org/commonjava/o11yphant/honeycomb/HoneycombFilter.java (hunk 1 above)
Comment (yma88) [reply honeycomb/core/src/main/java/org/commonjava/o11yphant/honeycomb/HoneycombFilter.java:72]: @ruhan1 I'll wrap in another PR

**Comment by yma88** [comment]
Close this, it's not necessary if honeycomb-core would not be an external dependence.

**Pull request closed without merging**